src/Common/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,124 @@
+        const _pages = createPageArr({ size, pageSize, selectedPageNo })
+        setPages(_pages)
+    }, [offset])
+
+    const selectPage = (selectedPageNo: number): void => {
+        const _pages = createPageArr({ size, pageSize, selectedPageNo })
+        setPages(_pages)
+        changePage(selectedPageNo)
+    }
+
+    const selectPreviousPage = (): void => {
+        const currPage = pages.find((page) => page.selected)
+        const selectedPageNo = currPage.value - 1
+        selectPage(selectedPageNo)
+    }
+
+    const selectNextPage = (): void => {
+        const currPage = pages.find((page) => page.selected)
+        const selectedPageNo = currPage.value + 1
+        selectPage(selectedPageNo)
+    }
+
+    const lastPageNo = offset + pageSize < size ? offset + pageSize : size
+    const pageNoIndex = pages.findIndex((page) => page.selected)
+    const visiblePages = pages.filter((page) => page.isVisible)
+    return (
+        <div className={rootClassName || ''}>
+            <div className="page-number">
+                {offset + 1} - {lastPageNo} of {size}
+            </div>
+
+            <ul className="pagination">
+                <li className="page">
+                    <button
+                        className="page__button page__button--icon"
+                        disabled={!pageNoIndex}
+                        onClick={selectPreviousPage}
+                        type="button"
+                    >
+                        <span className="left-icon">
+                            <i className="fa fa-chevron-left" />
+                        </span>
+                    </button>
+                </li>
+
+                {visiblePages.map((page) => (
+                    <PageValueItem
+                        value={page.value}
+                        isSelected={page.selected}
+                        key={page.value}
+                        selectPage={selectPage}
+                    />
+                ))}
+
+                <li className="page">
+                    <button
+                        className="page__button page__button--icon"
+                        disabled={pageNoIndex === pages.length - 1}
+                        onClick={selectNextPage}
+                        type="button"
+                    >
+                        <span className="left-icon">
+                            <i className="fa fa-chevron-right" />
+                        </span>
+                    </button>
+                </li>
+            </ul>
+
+            {!isPageSizeFix && (
+                <div className="select">
+                    <span className="select__placeholder">Rows per page</span>
+                    <PageSizeSelector
+                        pageSizeOptions={pageSizeOptions}
+                        pageSize={pageSize}
+                        changePageSize={changePageSize}
+                    />
+                </div>
```

Comment:
We can prefer using react select

Reply:
imported the logic from dashboard, even have left a note stating normalisation might benefit here.